frontend/pages/login.tsx

```diff
@@ -174,35 +185,31 @@ export default function Page() {
 	}
 
 	async function registerLogic() {
-		if (!form.current?.checkValidity()) {
-			return;
-		}
 		if (
-			passwordInput.current?.value === repeatPasswordInput.current?.value
+			!form.current?.checkValidity() ||
+			passwordInput.current?.value !== repeatPasswordInput.current?.value
 		) {
-			let url = "/api/register";
-			const response = await fetch(url, {
-				method: "POST",
-				headers: {
-					"Content-Type": "application/json",
-				},
-				body: JSON.stringify({
-					userEmail: emailInput.current?.value,
-					userName: nameInput.current?.value,
-					userPassword: passwordInput.current?.value,
-				}),
-			});
-			if (response.ok) {
-				toast.info("Account created!", { toastId: "accountToast" });
-				setLogin(true);
-			} else {
-				toast.error("Account already exists!", {
-					toastId: "accountToast",
-				});
-			}
+			return;
+		}
+		let url = "/api/register";
```

Comment:
```suggestion
		const url = "/api/register";
```

or just inline since you don't use `url` any time soon